src/psyclone/psyir/nodes/scoping_node.py

```diff
@@ -62,10 +62,16 @@ def __init__(self, children=None, parent=None, symbol_table=None):
         super(ScopingNode, self).__init__(self, children=children,
                                           parent=parent)
         if symbol_table:
-            self._symbol_table = symbol_table
+            if symbol_table._node is not None:
```

Comment:
Do we need `is not None`?

Reply:
Not in this case, but I still prefer it as mentioned in the second point in https://peps.python.org/pep-0008/#programming-recommendations

If we would have a node that could be evaluated true or false (probably a bad idea but anyway) the meaning of the condition would change if we didn't had the `is not None`